explos/channel-manager/teritori-register.go

```diff
@@ -0,0 +1,128 @@
+package main
+
+import (
+	"berty.tech/berty/v2/go/pkg/bertybot"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"math/rand"
+	"os"
+	"strings"
+)
+
+type teritoriData struct {
+	Step int               `json:"step"`
+	Data map[string]string `json:"data"`
+}
+
+func step0(ctx bertybot.Context, t teritoriData) {
+	if t.Data["pubkey"] == "" {
+		ctx.ReplyString("error: missing territoriPubKey")
+		return
+	}
+
+	// to modify
+	pubKey, err := os.ReadFile("public.key")
+	if err != nil {
+		ctx.ReplyString("error: " + err.Error())
+		return
+	}
+
+	nonce := rand.Int()
+
+	m, err := json.Marshal(teritoriData{
+		Step: 1,
+		Data: map[string]string{
+			"nonce": fmt.Sprintf("%d", nonce),
+			"sig":   base64.StdEncoding.EncodeToString(Sign((*[64]byte)(pubKey), []byte(fmt.Sprintf("%d%ssisi", nonce, t.Data["pubkey"])))),
+		},
+	})
```

Comment:
```suggestion
	data := teritoriData{
		Step: 1,
		Data: map[string]string{
			"nonce": fmt.Sprintf("%d", nonce),
			"sig":   base64.StdEncoding.EncodeToString(Sign((*[64]byte)(pubKey), []byte(fmt.Sprintf("%d%ssisi", nonce, t.Data["pubkey"])))),
		},
	}
	m, err := json.Marshal(data)
```

improve readability, and compiler debugging error messages.

Reply:
```suggestion
	proof := fmt.Sprintf("%d%ssisi", nonce, t.Data["pubkey"])
	sig := Sign((*[64]byte)(pubKey), []byte(proof))
	b64sig := base64.StdEncoding.EncodeToString(sig)
	nonceStr := fmt.Sprintf("%d", nonce)
	data := teritoriData{
		Step: 1,
		Data: map[string]string{
			"nonce": nonceStr,
			"sig":   b64sig,
		},
	}
	m, err := json.Marshal(data)
```

even better